sql/src/main/java/org/apache/druid/sql/calcite/expression/builtin/NestedDataOperatorConversions.java

```diff
@@ -207,7 +209,16 @@ public static class JsonKeysOperatorConversion implements SqlOperatorConversion
             )
         )
         .functionCategory(SqlFunctionCategory.USER_DEFINED_FUNCTION)
-        .returnTypeNullableArray(SqlTypeName.VARCHAR)
+        .returnTypeInference(
```

Comment:
`returnTypeNullableArray` behaves a bit different from this `returnTypeInference` because this is using `ReturnTypes.cascade` instead of `ReturnTypes.explicit`, and had some troubles with type mismatch nullable vs not until I made this change.

It might make more sense to make `returnTypeNullableArray` always be cascade nullable instead of always nullable, but I didn't make that change for now

Reply:
Hmm. If I read the code right, the difference is that `returnTypeNullableArray(SqlTypeName.VARCHAR)` yields `ARRAY<VARCHAR NULLABLE>` (nullable elements, non-nullable array), whereas, what you have here yields `ARRAY<VARCHAR NULLABLE> NULLABLE` (the array is itself nullable too).

Suggestion: rename `returnTypeNullableArray` to `returnTypeArrayWithNullableElements` and then introduce `returnTypeNullableArrayWithNullableElements` that does the thing you have here.